src/main/java/com/o19s/es/ltr/query/FeatureVectorQuery.java

```diff
@@ -0,0 +1,215 @@
+            return new ConstantScoreWeight(this) {
+                @Override
+                public Scorer scorer(LeafReaderContext context) throws IOException {
+                    return new ConstantScoreScorer(this, score(), DocIdSetIterator.all(context.reader().maxDoc()));
+                }
+            };
+        }
+
+        return new FVWeight(this);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!sameClassAs(obj)) {
+            return false;
+        }
+        FeatureVectorQuery that = (FeatureVectorQuery) obj;
+        return Objects.deepEquals(expression, that.expression);
+    }
+
+    @Override
+    public int hashCode() {
+    return 31 * classHash() + Objects.hash(expression);
+}
+
+    @Override
+    public String toString(String field) {
+        return "fv_query:"+field;
+    }
+
+    public class FVWeight extends FeatureVectorWeight {
+        protected FVWeight(Query query) {
+            super(query);
+        }
+
+        @Override
+        public void extractTerms(Set<Term> terms) {
+            // No-op
+        }
+
+        @Override
+        public float getValueForNormalization() throws IOException {
+            return 1.0f;
+        }
+
+        @Override
+        public Scorer scorer(LeafReaderContext context, Supplier<LtrRanker.FeatureVector> vectorSupplier) throws IOException {
+            Bindings bindings = new Bindings(){
+                @Override
+                public DoubleValuesSource getDoubleValuesSource(String name) {
+                    return new FVDoubleValuesSource(vectorSupplier, features.featureOrdinal(name));
+                }
+            };
+
+            DocIdSetIterator iterator = DocIdSetIterator.all(context.reader().maxDoc());
+            DoubleValuesSource src = expression.getDoubleValuesSource(bindings);
+            DoubleValues values = src.getValues(context, null);
+
+            return new DValScorer(this, iterator, values);
+        }
+
+        @Override
+        public Explanation explain(LeafReaderContext context, LtrRanker.FeatureVector vector, int doc) {
+            Suppliers.MutableSupplier<LtrRanker.FeatureVector> supplier = new Suppliers.MutableSupplier<>();
+            supplier.set(vector);
+
+            Bindings bindings = new Bindings(){
+                @Override
+                public DoubleValuesSource getDoubleValuesSource(String name) {
+                    return new FVDoubleValuesSource(supplier, features.featureOrdinal(name));
+                }
+            };
+
+            try {
+                DoubleValuesSource src = expression.getDoubleValuesSource(bindings);
+                DoubleValues values = src.getValues(context, null);
+                values.advanceExact(doc);
+                return Explanation.match((float) values.doubleValue(), "Evaluation of derived expression: " + expression.sourceText);
+            } catch (IOException ex) {
```

Comment:
feel free to update the signature on the base class and not catch IOException here, like you did for the scorer method